### PR TITLE
early_ta: use scattered array helpers

### PR DESCRIPTION
**Pull request by jenswi-linaro**
Simplifies the core linker script by replacing the hard coded
.rodata.early_ta section with use of SCATTERED_ARRAY_DEFINE_PG_ITEM()
instead.

Signed-off-by: Jens Wiklander <jens.wiklander@linaro.org>

### CODE REVIEW

**Review by jforissier — APPROVED** [review]
With or without my suggestion below:
`Reviewed-by: Jerome Forissier <jerome@forissier.org>`

File: scripts/ta_bin_to_c.py [hunk 1]
```diff
@@ -91,22 +102,11 @@ def main():
     f.write('\t\t\t')
     f.write(', '.join('0x' + csn[i:i + 2] for i in range(0, len(csn), 2)))
     f.write('\n\t\t},\n\t},\n')
-    f.write('\t.size = {:d},\n'.format(size))
+    f.write('\t.size = sizeof(ta_bin_' + ta_uuid.hex + '),\n')
```
[the remark this answers is not in the document]
Comment (jenswi-linaro) [reply scripts/ta_bin_to_c.py:105]: Sure

**Comment by jenswi-linaro** [comment]
There's no urgency to merge this before the release.

**Comment by jforissier** [comment]
> There's no urgency to merge this before the release.

My thoughts exactly ;-)

**Pull request merged**